src/msgs/measurements.rs

```diff
@@ -281,49 +413,470 @@ bitflags! {
 
 }
 
+// Used when DmtfMeasurementValueType::DeviceMode is set
 #[derive(Debug, Clone, PartialEq)]
 pub struct DeviceMode {
-    pub operational_mode_capabilities: OperationalMode,
-    pub operational_mode_state: OperationalMode,
-    pub device_mode_capabilities: DeviceModeActive,
-    pub device_mode_state: DeviceModeActive,
+    pub operational_mode_capabilities: OperationalModeBits,
+    pub operational_mode_state: OperationalModeBits,
+    pub device_mode_capabilities: DeviceModeBits,
+    pub device_mode_state: DeviceModeBits,
 }
 
 /// Table 45 of DMTF 1.2.0 spec
 #[derive(Debug, Clone, PartialEq)]
 pub struct DmtfMeasurement {
-    value_representation: DmtfMeasurementValueRepresentation,
-    value_type: DmtfMeasurementValueType,
+    pub value_representation: DmtfMeasurementValueRepresentation,
+    pub value_type: DmtfMeasurementValueType,
+    pub value: DmtfMeasurementValue,
+}
+
+#[derive(Debug)]
+pub enum ParseDmtfMeasurementError {
+    Representation(ParseDmtfMeasurementValueRepresentationError),
+    Type(ParseDmtfMeasurementValueTypeError),
+    Read(ReadError),
+    DigestTooLargeError,
+    BitStream(ParseBitStreamError),
+}
+
+impl From<ParseDmtfMeasurementValueTypeError> for ParseDmtfMeasurementError {
+    fn from(e: ParseDmtfMeasurementValueTypeError) -> Self {
+        ParseDmtfMeasurementError::Type(e)
+    }
+}
+
+impl From<ParseDmtfMeasurementValueRepresentationError>
+    for ParseDmtfMeasurementError
+{
+    fn from(e: ParseDmtfMeasurementValueRepresentationError) -> Self {
+        ParseDmtfMeasurementError::Representation(e)
+    }
+}
+
+impl From<ReadError> for ParseDmtfMeasurementError {
+    fn from(e: ReadError) -> Self {
+        ParseDmtfMeasurementError::Read(e)
+    }
+}
+
+impl From<DigestTooLargeError> for ParseDmtfMeasurementError {
+    fn from(_: DigestTooLargeError) -> Self {
+        ParseDmtfMeasurementError::DigestTooLargeError
+    }
+}
+
+impl From<ParseBitStreamError> for ParseDmtfMeasurementError {
+    fn from(e: ParseBitStreamError) -> Self {
+        ParseDmtfMeasurementError::BitStream(e)
+    }
+}
+
+impl DmtfMeasurement {
+    fn serialized_size(&self) -> usize {
+        3 + self.value.len()
+    }
+
+    fn write(&self, w: &mut Writer) -> Result<usize, BufferFullError> {
+        // Set bit 7 of byte 0
+        let mut byte0 = self.value_representation as u8;
+
+        // Set bits [6:0] of byte 0
+        byte0 |= self.value_type as u8;
+
+        w.put(byte0)?;
+        w.put_u16(self.value.len().try_into().unwrap())?;
```

Comment:
Could/should we use [`unwrap-lite`](https://github.com/oxidecomputer/hubris/tree/master/lib/unwrap-lite) to lower the codegen cost of unwraps? Or I could file an issue for this as I assume it affects more than just this PR?

Reply:
I created an issue for this. https://github.com/oxidecomputer/spdm/issues/44